python_modules/dagster/dagster/_core/storage/dagster_run.py

```diff
@@ -148,6 +148,34 @@ def __new__(
         )
 
 
+@whitelist_for_serdes
+class RootOpConcurrency(
```

Comment:
maybe RunRootOpConcurrency?

Reply:
I might just rename to `RunOpConcurrency`, and have it contain `root_key_counts`.

We don't have to leak the fact that we only care about root keys at the top level.